extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -538,6 +564,161 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     }
   }
 
+  /**
+   * Sends RTC request as specified by rtcConfig. Returns promise which
+   * resolves to the time that the RTC callout took to complete.
+   * If disableStalewhilerevalidate == true, then we will only
+   * send one RTC request per page. If it's !== true, then we always
+   * send two requests. The first will try
+   * to hit browser cache, and the second request will always bypass.
+   * This is a way for us to implement a simple stale-while-revalidate
+   * model.
+   * The targeting info from the RTC updates the targeting info on
+   * this object within mergeRtc.
+   * @return {?Promise<?Object>} An object of parameters to add to
+   *   the ad request url.
+   * @private
+   */
+  executeRtc_() {
+    if (rtcPromise) {
+      return this.mergeRtc();
+    }
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    if (!ampRtcPageElement) {
+      return Promise.resolve();
+    }
+    let endpoint;
+    rtcConfig = tryParseJson(ampRtcPageElement.textContent);
+    if (!isObject(rtcConfig) || !(endpoint = rtcConfig['endpoint']) ||
+        typeof endpoint != 'string' || !isSecureUrl(endpoint)) {
+      user().warn(TAG,
+          `invalid RTC config: ${ampRtcPageElement.textContent}`);
+      return Promise.resolve();
+    }
+    let rtcTotalTime;
+    const disableSWR = (
+        rtcConfig['disableStaleWhileRevalidate'] === true);
```

Comment:
Did you consider edge case values such as 'true', 1, and '1'?  Ideally there would be a helper function to ensure we are consistent across the codebase (perhaps one already exists?).  See sendAdRequestOnFailure you have below.

Also let's move this directly into the if condition below, no need to create variable.